Source/WebCore/style/StyleResolver.cpp

```diff
@@ -760,13 +760,15 @@ void Resolver::applyMatchedProperties(State& state, const MatchResult& matchResu
         m_matchedDeclarationsCache.add(style, parentStyle, cacheHash, matchResult);
 }
 
-void Resolver::setGlobalStateAfterApplyingProperties(const BuilderState& builderState)
+void Resolver::setGlobalStateAfterApplyingProperties(BuilderState& builderState)
 {
     // FIXME: This stuff should be somewhere else.
     for (auto& contentAttribute : builderState.registeredContentAttributes())
         ruleSets().mutableFeatures().registerContentAttribute(contentAttribute);
     if (builderState.style().usesViewportUnits())
         document().setHasStyleWithViewportUnits();
+
+    builderState.style().setUseSVGZoomRulesForLength(builderState.useSVGZoomRulesForLength());
```

Comment:
Why do this after applying properties? Anything that might need the used value during style building would get things wrong.

Reply:
this should be addressed here: https://github.com/WebKit/WebKit/pull/51643